gradle-plugins/model-compiler/src/main/java/io/spine/tools/compiler/enrichment/EnrichmentMap.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2018, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.compiler.enrichment;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimap;
+import com.google.protobuf.DescriptorProtos.DescriptorProto;
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto;
+import io.spine.option.RawListValue;
+import io.spine.type.TypeName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static io.spine.option.OptionsProto.enrichment;
+import static io.spine.option.OptionsProto.enrichmentFor;
+import static io.spine.option.RawListValue.getValueSeparator;
+
+/**
+ * Composes enrichment map for multiple message declarations.
+ *
+ * @author Alexander Litus
+ * @author Alex Tymchenko
+ * @author Alexander Yevsyukov
+ */
+class EnrichmentMap {
+
+    private static final String EMPTY_TYPE_NAME = "";
+
+    private final String packagePrefix;
+    private final TypeNameValue eventType;
+    private final TypeNameValue enrichmentType;
+
+    EnrichmentMap(String packagePrefix) {
+        this.packagePrefix = packagePrefix;
+        this.eventType = new TypeNameValue(enrichmentFor, packagePrefix);
+        this.enrichmentType = new TypeNameValue(enrichment, packagePrefix);
+    }
+
+    /**
+     * Obtains enrichment information found in the passed messages.
```

Comment:
In scope of clarifying things for the ease of writing the documentation, it would be good to describe what are the keys and values.